src/NzbDrone.Core/Download/Aggregation/Aggregators/AggregateLanguages.cs

```diff
@@ -76,9 +76,14 @@ public RemoteEpisode Aggregate(RemoteEpisode remoteEpisode)
                 languages = languages.Except(languagesToRemove).ToList();
             }
 
-            if ((languages.Count == 0 || (languages.Count == 1 && languages.First() == Language.Unknown)) && releaseInfo is { IndexerId: > 0 } && releaseInfo.Title.IsNotNullOrWhiteSpace())
+            if ((languages.Count == 0 || (languages.Count == 1 && languages.First() == Language.Unknown)) &&
+                releaseInfo?.Title?.IsNotNullOrWhiteSpace() == true &&
+                (releaseInfo is { IndexerId: > 0 } || releaseInfo.Indexer?.IsNotNullOrWhiteSpace() == true))
             {
-                var indexer = _indexerFactory.Get(releaseInfo.IndexerId);
+                var indexer = _indexerFactory.All()
+                    .FirstOrDefault(v =>
+                        (releaseInfo is { IndexerId: > 0 } && releaseInfo.IndexerId == v.Id) ||
+                        (releaseInfo.Indexer?.IsNotNullOrWhiteSpace() == true && v.Name.EqualsIgnoreCase(releaseInfo.Indexer)));
```

Comment:
With release/push can we end up with a scenario where the ID and the Name are for two different indexers?